macros/validation/validate_expression_metrics.sql

```diff
@@ -0,0 +1,13 @@
+{% macro validate_expression_metrics(full_set) %}
+
+    {# We loop through the full set here to ensure that metrics that aren't listed 
+    as expression are note dependent on another metric.  #}
+
+    {% for metric_name in full_set %}
+        {% set metric_relation = metric(metric_name)%}
+        {% if metric_relation.type != "expression" and metric_relation.metrics | length > 0 %}
+            {%- do exceptions.raise_compiler_error("The metric " ~ metric_relation.name ~ " was not an expression and dependent on another metric. This is not currently supported - if this metric depends on another metric, please change the type to expression.") %}
```

Comment:
```suggestion
            {%- do exceptions.raise_compiler_error("The metric " ~ metric_relation.name ~ " also references '" ~ metric_relation.metrics | join ("', ') ~ "' but its type is '" ~ metric_relation.type ~ "'. Only metrics of type 'expression' can reference other metrics.") %}
```

If I had a dollar for every thing I had said was not _currently_ supported that never became supported, I would not need to work here 😅

Reply:
How do we set this up and reap the infinite money from unsupported features that fall off roadmaps

Reply:
Fixed with:

```
{% set metric_relation_depends_on = metric_relation.metrics  | join (",") %}
{% if metric_relation.type != "expression" and metric_relation.metrics | length > 0 %}
     {%- do exceptions.raise_compiler_error("The metric " ~ metric_relation.name ~" also references '" ~ metric_relation_depends_on ~ "' but its type is '" ~ metric_relation.type ~ "'. Only metrics of type expression can reference other metrics.") %}
{%- endif %}
```